py3status/modules/do_not_disturb.py

```diff
@@ -145,7 +145,7 @@ class Py3status:
     thresholds = [(0, "bad"), (1, "good")]
 
     def post_config_hook(self):
-        servers = ["xfce4-notifyd", "dunst", "mako", None]
+        servers = ["dunst", "mako", "xfce4-notifyd", None]
```

Comment:
`servers = ["xfce4-notifyd", "mako", "dunst", "mako"]` due to `pgrep -f server`.

Reply:
can you explain please? we will try each of them in order and stop on the first match, no? and we want dunst to be tried first, and `xfce4-notifyd` last?

I also don't understand why we have `None` in the list...

Reply:
Don't remember. I looked at module.

`None` is there because of `for server in servers` + `break` on L140-145 and maybe for initializing a failed backend due to all missing servers. Can't be sure.

IIRC `pgrep -f server` is required to detect running `xfce4-notifyd`. Otherwise, it will force running `dunst` instead of `xfce4-notifyd`. Also, if `dunst` is running, we sometimes can detect that too. If nothing, we go with `dunst` anyway... so `None` is there for this one too.

Reply:
@ricci hey, are you still using `xfce4-notifyd` and `py3status`? Would you be able to test something for us?

I'm pretty sure we don't need `-f` flag for `pgrep` as it can lead to false positives. Could you please try `do_not_disturb` module where this line is set to

    def post_config_hook(self):
        servers = ["dunst", "mako", "xfce4-notifyd"]

And also could you please post the result of these commands?

```
$ pgrep -fl xfce4-notifyd
$ pgrep -fa xfce4-notifyd
$ pgrep -xl xfce4-notifyd
```

Basically, to make the precise search work, we must do `pgrep -x <result of pgrep -fl xfce4-notifyd>`, that's what I'd like to figure out.

Reply:
Yes, the latest version of the code in the pull request works for me with `xfce4-notifyd`. Here are the outputs of those commands:
```
> pgrep -fl xfce4-notifyd 
749 xfce4-notifyd
> pgrep -fa xfce4-notifyd
749 /usr/lib/xfce4/notifyd/xfce4-notifyd
 > pgrep -xl xfce4-notifyd
749 xfce4-notifyd
```
I will note that I'm running `xfce4-notifyd` via `systemcl --user start xfce4-notifyd.service`, which does start the process and keep it running continually. There may be some other way to set it to launch automatically via a socket or `dbus`, meaning that the process might not always be running (I don't know).

Reply:
Great, thank you very much! 

I think we should switch to the precise `pgrep -x` from `pgrep -f`, as the last test above shows that it works for `xfce4-notifyd`, and I think we need to set `servers = ["dunst", "mako", "xfce4-notifyd"]`